RR.App/Middleware/ExceptionHandlingMiddleware.cs

```diff
@@ -31,25 +31,32 @@ public async Task InvokeAsync(HttpContext context)
 
         private async Task HandleExceptionAsync(HttpContext context, Exception exception)
         {
-            var errorResponse = await GetRequestInfo(context, exception);
-            var message = "Internal Server Error. Please try again later.";
-
-            switch (exception)
+            try
             {
-                case CustomException:
-                    errorResponse.StatusCode = StatusCodes.Status404NotFound;
-                    message = exception.Message;
-                    break;
-                default:
-                    errorResponse.StatusCode = StatusCodes.Status500InternalServerError;
-                    _errorLoggingService.LogException(errorResponse);
-                    break;
-            }
+                var errorResponse = await GetRequestInfo(context, exception);
+                var message = "Internal Server Error. Please try again later.";
+
+                switch (exception)
+                {
+                    case CustomException:
+                        errorResponse.StatusCode = StatusCodes.Status404NotFound;
+                        message = exception.Message;
+                        break;
+                    default:
+                        errorResponse.StatusCode = StatusCodes.Status500InternalServerError;
+                        await _errorLoggingService.LogException(errorResponse);
+                        break;
+                }
 
-            context.Response.ContentType = "application/json";
-            context.Response.StatusCode = (int)errorResponse.StatusCode;
+                context.Response.ContentType = "application/json";
+                context.Response.StatusCode = (int)errorResponse.StatusCode;
 
-            await context.Response.WriteAsync(message);
+                await context.Response.WriteAsync(message);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
```

Comment:
Logic: Do we want the error to be displayed on the console?

Reply:
So this is a final fallback to view any errors on the azure console. This will or should never be hit, but in the event of something drastic occurring, we can view it on the console.

Reply:
@EduanO I understand. Perhaps then we should use something like this then:
```suggestion
                Console.ForegroundColor = ConsoleColor.Red;
                Console.Error.WriteLine(e);
                Console.ResetColor();
```